build.gradle

```diff
@@ -79,18 +79,20 @@ int lastPrevMinor = -1 // the minor version number from the prev major we most r
 for (String line : versionLines) {
   /* Note that this skips alphas and betas which is fine because they aren't
    * compatible with anything. */
-  Matcher match = line =~ /\W+public static final Version V_(\d+)_(\d+)_(\d+) .*/
+  Matcher match = line =~ /\W+public static final Version V_(\d+)_(\d+)_(\d+)(_beta\d+|_rc\d+)? .*/
   if (match.matches()) {
     int major = Integer.parseInt(match.group(1))
     int minor = Integer.parseInt(match.group(2))
     int bugfix = Integer.parseInt(match.group(3))
     Version foundVersion = new Version(major, minor, bugfix, false)
-    if (currentVersion != foundVersion) {
+    if (currentVersion != foundVersion
+        && (major == prevMajor || major == currentVersion.major)
+        && (versions.isEmpty() || versions.last() != foundVersion)) {
```

Comment:
I *think* this adds all the betas and rcs are wire and index compatible versions. I think, at least in the end, we want them to only be index and wire compatible if we only have only have them. Otherwise we'll be testing compatibility against betas while we're well into 6.0. Is this a thing we should handle in a followup?